src/background/index.ts

```diff
@@ -0,0 +1,68 @@
+console.log('Swarm Backend script started...')
+const beeApiUrl = 'http://localhost:1633'
+
+// typed in 'bzz://{content-address}' URI in search bar triggers redirect to gateway BZZ address
+// only react if google search has been set for searching
+chrome.webRequest.onBeforeRequest.addListener(
+  (details: chrome.webRequest.WebRequestBodyDetails) => {
+    console.log('Original BZZ Url', details.url)
+    const urlParams = new URLSearchParams(details.url)
+    for (const e of urlParams.entries()) {
+      console.log('entries', e)
+    }
+    const query = urlParams.get('oq')
+
+    if (!query || query.substr(0, 6) !== 'bzz://') return
+
+    console.log('bzz address', query)
+    console.log('redirect to', `${beeApiUrl}/bzz/${query.substr(6)}`)
+
+    return {
+      redirectUrl: `${beeApiUrl}/bzz/${query.substr(6)}`,
+    }
+  },
+  {
+    urls: ['https://www.google.com/search?*'],
+  },
+  ['blocking'],
+)
+
+//redirect from custom protocol
+chrome.webRequest.onBeforeRequest.addListener(
+  details => {
+    const urlArray = details.url.split('web%2Bbzz%3A%2F%2F')
+    const redirectUrl = `${beeApiUrl}/bzz/${urlArray[1]}`
+    console.log(`BZZ redirect to ${redirectUrl} from ${details.url}`)
+
+    if (urlArray.length > 1) {
```

Comment:
Isn't this condition always assumed to be true, since `urlArray[1]` is being accessed before?

If it is necessary, I'd probably go with an exit condition early on, so `urlArray[1]` is safe to use later.